components/mdx/mermaid.tsx

```
@@ -0,0 +1,170 @@
+    return (
+      <div className="rounded-lg border border-red-200 bg-red-50 p-4 text-red-700 dark:border-red-800 dark:bg-red-950 dark:text-red-300">
+        <p className="font-medium">Failed to render diagram</p>
+        <pre className="mt-2 overflow-auto text-sm">{error}</pre>
+      </div>
+    );
+  }
+
+  const diagramContent = (
+    <div
+      className="[&_svg]:mx-auto [&_svg]:max-w-full [&_svg]:h-auto [&_.node_rect]:rx-2"
+      dangerouslySetInnerHTML={{ __html: svg }}
```

Biome (2.3.14)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)